apps/angular/src/core/guards/authentication.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { inject } from '@angular/core';
+import { CanActivateFn, Router } from '@angular/router';
+
+import { map, tap } from 'rxjs';
+
+import { AuthorizationApiService } from '../services/authorization-api.service';
+import { RouterPaths } from '../model/router-paths';
+
+/**
+ * Guard checks if the user is logged in.
+ * @returns True if user is logged in and false otherwise.
+ */
+export const authorizationGuard: CanActivateFn = () => {
+	const authorizationService = inject(AuthorizationApiService);
+	const router = inject(Router);
+
+	return authorizationService.isAuthenticated().pipe(
+		tap(isAuthenticated => {
+			if (!isAuthenticated) {
+				router.navigate([RouterPaths.Login]);
+			}
+		}),
+		map(isAuthenticated => isAuthenticated),
```

Comment:
You don't need to use `router.navigate` as side effect. You can just do like this
```suggestion
		 map(isAuthenticated => {
			if (!isAuthenticated) {
				return router.parseUrl(RouterPaths.Login);
			}
			return true;
		}),
```